### ISSUE TITLE
Missing NDC Codes

### ISSUE THREAD

**Issue by cukarthik** [OHDSI/Vocabulary-v5.0#523]
I'm using [CanMed](https://seer.cancer.gov/oncologytoolbox/canmed/ndconc/) from NCI, and the below are NDC (11-digit) codes that I could not find in the OMOP vocabulary. Some of these match on the 9-digit NDC code, but I'm only able to match 8 more codes. @cgreich , I'm not sure if this is a real issue or not, but wanted to post to get your opinion on how to get these codes into the vocabulary.

NDC-11 (Package) | NDC-9 (Product) | Generic Name | Brand Name
-- | -- | -- | --
00078066984 | 00078-0669 | Ofatumumab | Arzerra
45963061456 | 45963-0614 | Irinotecan Hydrochloride | Irinotecan Hydrochloride
53964000100 | 53964-0001 | Thiotepa | Tepadina
53964000201 | 53964-0002 | Thiotepa | Tepadina
59115014105 | 59115-0141 | Prednisone | Prednisone
00310060030 | 00310-0600 | Tamoxifen Citrate | Nolvadex
00310060460 | 00310-0604 | Tamoxifen Citrate | Nolvadex
50111082376 | 50111-0823 | Ondansetron | Ondansetron
50111082478 | 50111-0824 | Ondansetron | Ondansetron
55390014801 | 55390-0148 | Granisetron Hydrochloride | Granisetron Hydrochloride
55390014810 | 55390-0148 | Granisetron Hydrochloride | Granisetron Hydrochloride
60505036100 | 60505-0361 | Prednisolone | Prednisolone
60505036200 | 60505-0362 | Prednisolone | Prednisolone
60505036201 | 60505-0362 | Prednisolone | Prednisolone
60505271000 | 60505-2710 | Granisetron Hydrochloride | Granisetron Hydrochloride
60505271002 | 60505-2710 | Granisetron Hydrochloride | Granisetron Hydrochloride
60505271008 | 60505-2710 | Granisetron Hydrochloride | Granisetron Hydrochloride
6332301020 | 63323-0102 | Gemcitabine Hydrochloride | Gemcitabine Hydrochloride
6332310213 | 63323-0102 | Gemcitabine Hydrochloride | Gemcitabine Hydrochloride
6332310294 | 63323-0102 | Gemcitabine Hydrochloride | Gemcitabine Hydrochloride
68788182301 | 68788-1823 | Ondansetron Hydrochloride | Ondansetron Hydrochloride

**Comment by cgreich** [comment]
@cukarthik:

Yeah. We need to figure out something with the NDCs and their 9 and 11 digit versions. In principle, you only need the 9-digit version, since the remaining 2 digits ought to contain identical products but different packaging. But we don't have it set up like that.

In your first example it works fine:

ndc_code | ndc_class | ndc_name | rx_id | rx_class | rx_name
-- | -- | -- | -- | -- | --
000780669 | 9-digit NDC | ofatumumab 20mg/mL INTRAVENOUS INJECTION, SOLUTION | 46275167 | Quant Branded Drug | 5 ML ofatumumab 20 MG/ML Injection [Arzerra]
00078066913 | 11-digit NDC | 5 ML ofatumumab 20 MG/ML Injection [Arzerra] | 46275167 | Quant Branded Drug | 5 ML ofatumumab 20 MG/ML Injection [Arzerra]
00078066961 | 11-digit NDC | 5 ML ofatumumab 20 MG/ML Injection [Arzerra] | 46275167 | Quant Branded Drug | 5 ML ofatumumab 20 MG/ML Injection [Arzerra]

**Comment by cukarthik** [comment]
Yea, I noticed the inconsistency b/w 9 and 11 digits. I wanted to leverage the groupings that NCI create in CanMed, but can only do it if the NDC codes are complete. The other thing we've tried is to do a general string match on the Generic Name column in CanMed with Ingredient concepts, but that's not perfect either.

**Comment by cgreich** [comment]
If you think Canmed is a valid and reliable source we could just incorporate it. Do you?

**Comment by cukarthik** [comment]
I believe so. It is something that is evolving (i.e. the most recent version has about 2K rows that the one we downloaded a couple of years before). They also have a list of cancer related procedure codes ([HCPS](https://seer.cancer.gov/oncologytoolbox/canmed/hcpcs/)) as well that map to SEER Rx Categories.

**Comment by Alexdavv** [comment]
> For Ganisetron we only have NDC 605052711 instead of 605052710. The last digit of the product is different, could be the concentration (not in your table). We don't have any Prednisolone with a similar concept_code. Can you get more about these?

I'm not saying this is a good reason why we're missing them, but they are outdated in 2008:
![image](https://user-images.githubusercontent.com/32357692/132473747-f8c5a4f6-bb75-4017-a506-a532c974700b.png)

> If you think Canmed is a valid and reliable source we could just incorporate it. Do you?
Given the fact the official FDA NDC sources are missing some historical stuff, we need to consider doing so.

**Comment by cukarthik** [comment]
Besides the missing NDC codes, it would be nice to be able to match the GenericName column (which is the nonproprietary name field in NDC) to a RxNorm Ingredient.

**Comment by cgreich** [comment]
That's the least common denominator solution, @cukarthik

**Comment by MeliHenry** [comment]
@TinyRickC137 do we have an update on this?

**Comment by TinyRickC137** [comment]
@MeliHenry Not figured out yet. 
This issue stays as one of the reminders regarding the incorporation of other NDC sources, that are not in use today. Let's keep it for now until we are ready
